File: Clase_01/Practica_01/Comparable.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Practica_01
{
    // EJERCICIO 1 🐸
    //Nota: Asumiremos que el elemento que recibe el mensaje y el objeto recibido por parámetro
    //son del “mismo tipo”.

    /// <summary>
    /// Representa cualquier objeto que sabe comparase con otro objeto de su “mismo tipo”.
    /// </summary>
    public interface Comparable
    {
        /// <summary>
        /// Describe true si el objeto que recibe el mensaje es el mismo que el “comparable” recibido por parámetro, 
        /// describe false en caso contrario.
        /// </summary>
        /// <returns>Bool</returns>
        bool sosIgual(Comparable comparable);

        /// <summary>
        /// Describe true si el objeto que recibe el mensaje es más chico que el “comparable” recibido por parámetro; 
        /// describe false en caso contrario.
        /// </summary>
        /// <returns>Bool</returns>
        bool sosMenor(Comparable comparable);

        /// <summary>
        /// Describe true si el objeto que recibe el mensaje es más chico que el "comparable" recibido por parámetro;
        /// describe false en caso contrario.
        /// </summary>
        /// <returns>Bool</returns>
        bool sosMayor(Comparable comparable);

    }
}

File: interfaces/src/main/java/comparable/Comparable.java
package comparable;

/**
 * Interfaz que define un contrato para comparar objetos.
 */
public interface Comparable {

    /**
     * Determina si el objeto que recibe el mensaje es igual al objeto
     * especificado.
     *
     * @param comparable el objeto a comparar.
     * @return {@code true} si ambos objetos son iguales; {@code false} en caso contrario.
     */
    Boolean sosIgual(Comparable comparable);

    /**
     * Determina si el objeto que recibe el mensaje es menor que el objeto
     * especificado.
     *
     * @param comparable el objeto a comparar.
     * @return {@code true} si el objeto receptor es menor; {@code false} en caso contrario.
     */
    Boolean sosMenor(Comparable comparable);

    /**
     * Determina si el objeto que recibe el mensaje es mayor que el objeto
     * especificado.
     *
     * @param comparable el objeto a comparar.
     * @return {@code true} si el objeto receptor es mayor; {@code false} en caso contrario.
     */
    Boolean sosMayor(Comparable comparable);
}